internal/pkg/service/stream/storage/node/coordinator/slicerotation/operator.go

```diff
@@ -248,38 +256,62 @@ func (o *operator) rotateSlice(ctx context.Context, slice *sliceData) {
 
 	// Rotate slice
 	o.logger.Infof(ctx, "rotating slice for upload: %s", cause)
+	err = o.rotateSliceWithState(ctx, slice)
+	if err != nil {
+		return
+	}
```

Comment:
I think you mean this code.
https://github.com/keboola/keboola-as-code/blob/fix-node-operator-logging-consolidation/internal/pkg/service/stream/storage/node/coordinator/slicerotation/operator.go#L281-L290
If you can see, we can receive error when switching state to `close`.
In case we get error of this switching operation we try to increment the RetryAttempt. However, I use another error `iErr` to differentiate between these two.
1. In case we cannot increment attempt, it is  `!processed`
2. In case we can switch the increment attempt the function returns `nil` which set's it to `processed`.
3. In case we have no error is set's it to `processed`.

I've had different refactor in place but I decided to not go with it as your code was approved.

Reply:
ping @Matovidlo

Reply:
should be fixed https://github.com/keboola/keboola-as-code/pull/1960/commits/454749fc6caec2df4d38fd6cc05a96390933d3d4